spec/index.spec.js

```diff
@@ -52,11 +52,35 @@ describe('HtmlWebpackIncludeAssetsPlugin', function () {
       done();
     });
 
-    it('should throw an error if any of the asset options are not strings', function (done) {
+    it('should throw an error if any of the asset options are not strings or objects', function (done) {
       var theFunction = function () {
         return new HtmlWebpackIncludeAssetsPlugin({ assets: ['foo.js', true, 'bar.css'], append: false });
       };
-      expect(theFunction).toThrowError(/(options assets key array should not contain non-strings)/);
+      expect(theFunction).toThrowError(/(options assets key array must contain only strings and objects)/);
+      done();
+    });
+
+    it('should throw an error if any of the asset options are objects missing the path property', function (done) {
+      var theFunction = function () {
+        return new HtmlWebpackIncludeAssetsPlugin({ assets: ['foo.js', { type: 'js' }, 'bar.css'], append: false });
+      };
+      expect(theFunction).toThrowError(/(options assets key array objects must contain a string path property)/);
+      done();
+    });
+
+    it('should throw an error if any of the asset options are objects missing the type property', function (done) {
+      var theFunction = function () {
+        return new HtmlWebpackIncludeAssetsPlugin({ assets: ['foo.js', { path: 'baz.js' }, 'bar.css'], append: false });
+      };
+      expect(theFunction).toThrowError(/(options assets key array objects must contain a string type property)/);
+      done();
+    });
+
+    it('should throw an error if any of the asset options are objects with an invalid type property', function (done) {
+      var theFunction = function () {
+        return new HtmlWebpackIncludeAssetsPlugin({ assets: ['foo.js', { path: 'baz.js', type: 'foo' }, 'bar.css'], append: false });
+      };
+      expect(theFunction).toThrowError(/(options assets key array objects must contain a string type property)/);
       done();
```

Comment:
maybe add a test to make sure it throws an error for non-string path? (ex: `{ path: 123, type: 'js' } `)